backend/main.py

```
@@ -0,0 +1,227 @@
+        # Add AI response to history
+        ai_msg = {
+            "content": ai_response,
+            "type": "assistant",
+            "timestamp": datetime.now()
+        }
+        chat_history.append(ai_msg)
+        
+        return response
+        
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error processing chat: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)